cuda_core/tests/test_linker.py

```diff
@@ -40,6 +44,19 @@ def compile_ltoir_functions(init_cuda):
     return object_code_a_ltoir, object_code_b_ltoir, object_code_c_ltoir
 
 
+@contextmanager
+def skip_version_specific_linker_options():
+    if culink_backend:
+        return
+    try:
+        yield
+    except nvjitlink.nvJitLinkError as e:
+        if e.status == nvjitlink.Result.ERROR_UNRECOGNIZED_OPTION:
+            pytest.skip("current nvjitlink version does not support the option provided")
```

Comment:
1. this probably needs a reraise when the error code is not unrecognized option:
```suggestion
    except nvjitlink.nvJitLinkError as e:
        if e.status == nvjitlink.Result.ERROR_UNRECOGNIZED_OPTION:
            pytest.skip("current nvjitlink version does not support the option provided")
        else:
            raise
```
2. I am wondering if this check should be moved to `_exception_manager()` in `_linker.py`? WDYT?

Reply:
I have separate reviews for improved handling of special exceptions in the linker and program classes, however this check is unique to our testsuite and not the linker itself. It just allows us to test options such as "no_cache" (and otherwhich aren't supported on nvjitlink 12.4. The same style of exception based skip is implemented in the kernel attribute review. This way we don't have to keep track of which options are supported in which CTK

In the other reviews I've added lines to the existing exception manager which provide informative error messages based on the error code of the exception raised by check_result. That way we can provide suggested solutions or whatever else.